src/append/journald/field.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2024 FastLabs Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This field is derived from https://github.com/swsnr/systemd-journal-logger.rs/blob/v2.2.0/src/fields.rs.
+
+//! Write well-formated journal fields to buffers.
+
+use std::fmt::Arguments;
+use std::io::Write;
+
+use log::kv::Value;
+
+pub(super) enum FieldName<'a> {
+    WellFormed(&'a str),
+    WriteEscaped(&'a str),
+}
+
+/// Whether `c` is a valid character in the key of a journal field.
+///
+/// Journal field keys may only contain ASCII uppercase letters A to Z,
+/// numbers 0 to 9 and the underscore.
+fn is_valid_key_char(c: char) -> bool {
+    matches!(c, 'A'..='Z' | '0'..='9' | '_')
+}
+
+/// Write an escaped `key` for use in a systemd journal field.
+///
+/// See [`super::Journald`] for the rules.
+fn write_escaped_key(key: &str, buffer: &mut Vec<u8>) {
+    // Key length is limited to 64 bytes
+    let mut remaining = 64;
+
+    let escaped = key
+        .to_ascii_uppercase()
+        .replace(|c| !is_valid_key_char(c), "_");
+
+    if escaped.starts_with(|c: char| matches!(c, '_' | '0'..='9')) {
+        buffer.extend_from_slice(b"ESCAPED_");
+        remaining -= 8;
+    }
+
+    for b in escaped.into_bytes() {
+        if remaining == 0 {
+            break;
+        }
+        buffer.push(b);
+        remaining -= 1;
+    }
+}
```

Comment:
@swsnr This is a solution for your leaving TODO.

Sorry I don't directly depend your crate mainly due to this crate is also a logger implementation and I only need part of your code. The credit and origin are documented, though.